dom.go

```diff
@@ -2,9 +2,65 @@ package vecty
 
 import "github.com/gopherjs/gopherjs/js"
 
+// Renderable is a user-defined renderable component.
+type Renderable interface {
+	Component
+	Render() Component
+}
+
+// Core is the struct which all components embed.
+type Core struct {
+	component Renderable
+	body      Component
+}
+
+// Apply implements the Markup interface.
+func (c *Core) Apply(e *Element) {
+	e.AddChild(c.component)
+}
+
+// Unmount implements the Component interface.
+func (c *Core) Unmount() {
+	// Break the circular link between c (Core) and c.component (Renderable) or
+	// else one object could not be garbage collector without the other (both
+	// would have to be unused).
+	c.component = nil
+	c.body.Unmount()
+}
+
+// Node implements the Component interface.
+func (c *Core) Node() *js.Object {
+	return c.body.Node()
+}
+
+// Reconcile implements the Component interface.
+func (c *Core) Reconcile(oldComp Component) {
```

Comment:
> just because `reflect` seems cheaty / nasty.. but I'd be interested to hear unbiased opinions? @shurcooL can give thoughts too maybe?

I don't know, it doesn't seem that bad to me. It looks like a case where reflect might be a right tool.

Is it guaranteed to have `*Core` or can you only expect that to be the case due to convention?

If there's something to be opposed to, I think it might be the higher level considerations, but IMO it's not the use of reflect for that specific task.